lib/rouge/lexers/openedge.rb

```diff
@@ -0,0 +1,399 @@
+VALUES VAR VARIABLE VERBOSE VERSION VERT VERTICAL VIEW VIEW-AS
+VIEW-FIRST-COLUMN-ON-REOPEN VIRTUAL-HEIGHT VIRTUAL-HEIGHT-CHARS VIRTUAL-HEIGHT-P
+VIRTUAL-HEIGHT-PIXELS VIRTUAL-WIDTH VIRTUAL-WIDTH-CHARS VIRTUAL-WIDTH-P
+VIRTUAL-WIDTH-PIXELS VISIBLE VOID WAIT WAIT-FOR WARNING WEB-CONTEXT WEEKDAY WHEN
+WHERE WHILE WIDGET WIDGET-E WIDGET-ENTER WIDGET-ID WIDGET-L WIDGET-LEAVE
+WIDGET-POOL WIDTH WIDTH WIDTH-CHARS WIDTH-P WIDTH-PIXELS WINDOW WINDOW-MAXIM
+WINDOW-MAXIMIZED WINDOW-MINIM WINDOW-MINIMIZED WINDOW-NAME WINDOW-NORMAL
+WINDOW-STA WINDOW-STATE WINDOW-SYSTEM WITH WORD-INDEX WORD-WRAP
+WORK-AREA-HEIGHT-PIXELS WORK-AREA-WIDTH-PIXELS WORK-AREA-X WORK-AREA-Y WORKFILE
+WORK-TAB WORK-TABLE WRITE WRITE-CDATA WRITE-CHARACTERS WRITE-COMMENT
+WRITE-DATA-ELEMENT WRITE-EMPTY-ELEMENT WRITE-ENTITY-REF WRITE-EXTERNAL-DTD
+WRITE-FRAGMENT WRITE-JSON WRITE-MESSAGE WRITE-PROCESSING-INSTRUCTION
+WRITE-STATUS WRITE-XML WRITE-XMLSCHEMA X XCODE XML-DATA-TYPE
+XML-ENTITY-EXPANSION-LIMIT XML-NODE-TYPE XML-SCHEMA-PATH
+XML-STRICT-ENTITY-RESOLUTION XML-SUPPRESS-NAMESPACE-PROCESSING X-OF XREF
+XREF-XML Y YEAR YEAR-OFFSET YES YES-NO YES-NO-CANCEL Y-OF
+        )
+      end
+
+      def self.keywords_type
+        @keywords_type ||= Set.new %w(
+BLOB CHARACTER CHAR CLOB COM-HANDLE DATE DATETIME DATETIME-TZ DECIMAL DEC 
+HANDLE INT64 INTEGER INT LOGICAL LONGCHAR MEMPTR RAW RECID ROWID WIDGET-HANDLE          
+        )
+      end
+
+      start { push :bol }
+
+      state :expr_bol do
+        mixin :inline_whitespace
+
+        rule(//) { pop! }
+      end
+
+      # :expr_bol is the same as :bol but without labels, since
+      # labels can only appear at the beginning of a statement.
+      state :bol do
+        rule /#{id}:(?!:)/, Name::Label
+        mixin :expr_bol
+      end
+
+      state :inline_whitespace do
+        rule /[ \t\r]+/, Text
+        rule /\\\n/, Text # line continuation
+        rule %r(/(\\\n)?[*].*?[*](\\\n)?/)m, Comment::Multiline
+      end
+
+      state :whitespace do
+        rule /\n+/m, Text, :bol
+        rule %r(//(\\.|.)*?$), Comment::Single, :bol
+        mixin :inline_whitespace
+      end
+
+      state :expr_whitespace do
+        rule /\n+/m, Text, :expr_bol
+        mixin :whitespace
+      end
+
+      state :statements do
+        mixin :whitespace
+        rule /(u8|u|U|L)?"/, Str, :string
+        rule %r((u8|u|U|L)?'(\\.|\\[0-7]{1,3}|\\x[a-f0-9]{1,2}|[^\\'\n])')i, Str::Char
+        rule %r((\d+[.]\d*|[.]?\d+)e[+-]?\d+[lu]*)i, Num::Float
+        rule %r(\d+e[+-]?\d+[lu]*)i, Num::Float
+        rule /0x[0-9a-f]+[lu]*/i, Num::Hex
+        rule /0[0-7]+[lu]*/i, Num::Oct
+        rule /\d+[lu]*/i, Num::Integer
+        rule %r(\*/), Error
+        rule %r([~!%^*+=\|?:<>/-]), Operator
+        rule /[()\[\],]/, Punctuation
+        rule /(?:true|false|NULL)\b/, Name::Builtin
+        rule id do |m|
+          name = m[0]
+
+          if self.class.keywords.any?{ |s| s.casecmp(name) == 0}
```

Comment:
As of Ruby 2.6.3, `String#casecmp` is just comparing strings regardless of case with no Unicode case folding (compare that with [`String#casecmp?`](https://ruby-doc.org/core-2.6.3/String.html#method-i-casecmp-3F)). Given that, it'd be better to call `String#upcase` in the assignment to `name`.